state/charmref.go

```diff
@@ -144,13 +135,12 @@ func finalAppCharmRemoveOps(appName string, curl *string) []txn.Op {
 }
 
 // charmDestroyOps implements the logic of charm.Destroy.
-func charmDestroyOps(st modelBackend, curl *charm.URL) ([]txn.Op, error) {
+func charmDestroyOps(st modelBackend, curl *string) ([]txn.Op, error) {
 	db := st.db()
 	charms, cCloser := db.GetCollection(charmsC)
 	defer cCloser()
 
-	charmKey := curl.String()
-	charmOp, err := nsLife.destroyOp(charms, charmKey, nil)
+	charmOp, err := nsLife.destroyOp(charms, *curl, nil)
```

Comment:
if we are passing around strptr everywhere, we need nil checks, otherwise just pass by value

Reply:
This comment is relevant to pr #14174, will Resolve there.